sycl/doc/extensions/proposed/sycl_ext_oneapi_syclbin.asciidoc

```diff
@@ -0,0 +1,230 @@
+Copyright (C) 2025 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 9 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+This extension also depends on the following other SYCL extensions:
+
+* link:../experimental/sycl_ext_oneapi_kernel_compiler.asciidoc[
+  sycl_ext_oneapi_kernel_compiler]
+* link:../experimental/sycl_ext_oneapi_properties.asciidoc[
+  sycl_ext_oneapi_properties]
+
+
+== Status
+
+This is a proposed extension specification, intended to gather community
+feedback.  Interfaces defined in this specification may not be implemented yet
+or may be in a preliminary state.  The specification itself may also change in
+incompatible ways before it is finalized.  *Shipping software products should
+not rely on APIs defined in this specification.*
+
+
+== Overview
+
+This extensions adds APIs, built upon the existing SYCL 2020 `kernel_bundle`
+APIs, for loading precompiled "SYCLBIN" files. Using these, SYCL code can
+dynamically load kernel binaries produced by the associated compiler, the format
+and options for which are defined by the SYCL implementation.
+
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_SYCLBIN` to one of the values defined in the table
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+
+=== New kernel_bundle interfaces
+
+|====
+a|
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
+namespace sycl::ext::oneapi::experimental {
+
+template<bundle_state State, typename PropertyListT = empty_properties_t>
+kernel_bundle<State> get_kernel_bundle(const context& ctxt,
+                                       const std::vector<device>& devs,
+                                       const sycl::span<char>& bytes,
+                                       PropertyListT props = {})
+
+#if __cplusplus >= 202002L                                
+template<bundle_state State, typename PropertyListT = empty_properties_t>
+kernel_bundle<State> get_kernel_bundle(const context& ctxt,
+                                       const std::vector<device>& devs,
+                                       const std::span<char>& bytes,
+                                       PropertyListT props = {}) (1)
```

Comment:
When numbering functions in a synopsis, always number them all.  So, number the first (1) and the second (2), even if you don't call out (1) in the description.

My preferred style is to put the number on the first line of the function's definition, indented two columns beyond the longest line in the definition.  In addition, align all the numbers, so they start in the same column.  Thus, the numbers end up indented two columns beyond the longest line in any definition in the synopsis.

Reply:
I suspect I've addressed this. Please let me know if I misunderstood.